src/openapi/streaming/streaming.js

```diff
@@ -35,35 +35,48 @@ const ignoreSubscriptions = {};
  * then follows the Connection state model.
  */
 function init() {
-    setNewContextId.call(this);
-
     this.connection = new Connection(this.options, this.baseUrl, this.transport);
 
-    updateConnectionQuery.call(this);
-
     this.connection.setStateChangedCallback(onConnectionStateChanged.bind(this));
+    this.connection.setUnauthorizedCallback(onUnauthorized.bind(this));
     this.connection.setReceivedCallback(onReceived.bind(this));
     this.connection.setConnectionSlowCallback(onConnectionSlow.bind(this));
     this.connection.setErrorCallback(onErrorCallback.bind(this));
 
     // start the connection process
-    this.connection.start(onConnectionStarted.bind(this));
+    connect.call(this);
+}
+
+/**
+ * The streaming connection received a unauthorized - the token is
+ * being rejected so we should get a new one.
+ */
+function onUnauthorized() {
+    this.authProvider.tokenRejected();
 }
 
 /**
  * Reconnects the streaming socket when it is disconnected
  */
-function reconnect() {
-    if (this.connectionState !== this.CONNECTION_STATE_DISCONNECTED) {
-        log.error(LOG_AREA, 'Only call reconnect on a disconnected streaming connection');
+function connect() {
+    if (this.connectionState !== this.CONNECTION_STATE_DISCONNECTED && this.connectionState !== this.CONNECTION_STATE_INITIALIZING) {
+        log.error(LOG_AREA, 'Only call connect on a disconnected streaming connection');
         return;
     }
 
-    setNewContextId.call(this);
-    updateConnectionQuery.call(this);
+    const startConnection = () => {
+        setNewContextId.call(this);
+        updateConnectionQuery.call(this);
 
-    this.reconnecting = true;
-    this.connection.start(onConnectionStarted.bind(this));
+        this.connection.start(onConnectionStarted.bind(this));
+    };
+
+    const expiry = this.authProvider.getExpiry();
+    if (expiry && expiry < Date.now()) {

```

Comment:
In which scenarios expiry might be missing? Do we want to connect in such scenarios?

Reply:
Not for saxotrader, but I think we allow other users of the library to just pass in a token.

Reply:
It seems we require an expiry, I will rmeove the if

Reply:
*remove the `expiry &&` part